Core/Utilities/ClassAd/ClassAdLight.py

```diff
@@ -114,144 +110,142 @@ def insertAttributeBool( self, name, attribute ):
-    
-  def insertAttributeVectorStringList( self, name, attributelist ):
-    """Insert a named list of string lists 
+
+  def insertAttributeVectorStringList(self, name, attributelist):
+    """Insert a named list of string lists
     """
 
     listOfLists = []
     for stringList in attributelist:
-      #tmp = map ( lambda x : '"' + x + '"', stringList )
-      tmpstr = ','.join( stringList )
+      # tmp = map ( lambda x : '"' + x + '"', stringList )
+      tmpstr = ','.join(stringList)
       listOfLists.append('{' + tmpstr + '}')
-    self.contents[name] = '{' + ','.join(listOfLists) + '}'    
+    self.contents[name] = '{' + ','.join(listOfLists) + '}'
 
-  def lookupAttribute( self, name ):
+  def lookupAttribute(self, name):
     """Check the presence of the given attribute
     """
 
-    return self.contents.has_key( name )
+    return name in self.contents
 
-  def set_expression( self, name, attribute ):
+  def set_expression(self, name, attribute):
     """Insert a named expression attribute
     """
 
-    self.contents[name] = str( attribute )
+    self.contents[name] = str(attribute)
 
-  def get_expression( self, name ):
+  def get_expression(self, name):
     """Get expression corresponding to a named attribute
     """
 
-    if self.contents.has_key( name ):
-      if isinstance( self.contents[name], ( int, long ) ):
-        return str( self.contents[name] )
-      else :
-        return self.contents[name]
-    else:
-      return ""
+    if name in self.contents:
+      if isinstance(self.contents[name], (int, long)):
+        return str(self.contents[name])
+      return self.contents[name]
+    return ""
 
-  def isAttributeList( self, name ):
+  def isAttributeList(self, name):
     """ Check if the given attribute is of the List type
     """
-    attribute = self.get_expression( name ).strip()
-    return attribute.startswith( '{' )
+    attribute = self.get_expression(name).strip()
+    return attribute.startswith('{')
 
-  def getListFromExpression( self, name ):
+  def getListFromExpression(self, name):
     """ Get a list of strings from a given expression
     """
 
-    tempString = self.get_expression( name ).strip()
+    tempString = self.get_expression(name).strip()
     listMode = False
     if tempString.startswith('{'):
       tempString = tempString[1:-1]
       listMode = True
-      
-    tempString = tempString.replace( " ", "" ).replace( '\n','' )  
+
+    resultList = []
+    tempString = tempString.replace(" ", "").replace('\n', '')
     if tempString.find('{') < 0:
       if not listMode:
-        tempString = tempString.replace( "\"", "" )
-        return tempString.split( ',' ) 
-    
-    resultList = []    
-    while tempString:      
-      if tempString.find( '{' ) == 0 :
-        end = tempString.find( '}' )
-        resultList.append(tempString[:end+1])
-        tempString = tempString[end+1:]
+        tempString = tempString.replace("\"", "")
+        if tempString:
```

Comment:
The only real fix is here. This I believe is a bug. It was manifesting specifically because of this change: https://github.com/DIRACGrid/DIRAC/pull/3875/files?utf8=%E2%9C%93&diff=unified&w=1 and specifically when calling `platformList = classAdJob.getListFromExpression('Platform')` at line 1249 --- the result, for empty platform, was `['']`

Reply:
Yes, this is a bug. I think this can be simpler fixed by returning [] right after the first line in this method:
if not tempString: return []

Reply:
Done. Please check latest commit.